Source/ObjectiveDropboxOfficial/Shared/Handwritten/Networking/DBDelegate.m

```diff
@@ -330,10 +335,15 @@ - (NSString *)sessionIdWithSession:(NSURLSession *)session {
 
 - (DBSessionData *)sessionDataWithSession:(NSURLSession *)session {
   NSString *sessionId = [self sessionIdWithSession:session];
-  if (!_sessionData[sessionId]) {
-    _sessionData[sessionId] = [[DBSessionData alloc] initWithSessionId:sessionId];
+  DBSessionData *sessionData = self.sessionData[sessionId];
+  if (sessionData == nil) {
+    sessionData = [[DBSessionData alloc] initWithSessionId:sessionId];
+    if (sessionData != nil) {
```

Comment:
Redundant check?

Reply:
if it fails should it be allowed to wipe out what's there?

Reply:
shadowing the sessionData property makes this more confusing.